src/main/java/jenkins/plugins/slack/webhook/WebhookEndpoint.java

```diff
@@ -0,0 +1,250 @@
+import hudson.model.UnprotectedRootAction;
+
+import hudson.security.ACL;
+
+import javax.servlet.ServletException;
+
+import java.io.IOException;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import java.util.logging.Logger;
+
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import jenkins.plugins.slack.webhook.model.JsonResponse;
+import jenkins.plugins.slack.webhook.model.SlackPostData;
+import jenkins.plugins.slack.webhook.model.SlackTextMessage;
+import jenkins.plugins.slack.webhook.model.SlackWebhookCause;
+
+import jenkins.plugins.slack.webhook.exception.CommandRouterException;
+import jenkins.plugins.slack.webhook.exception.RouteNotFoundException;
+
+
+
+
+@Extension
+public class WebhookEndpoint implements UnprotectedRootAction {
+
+    private String slackUser;
+
+    private GlobalConfig globalConfig;
+
+    private static final Logger LOGGER =
+        Logger.getLogger(WebhookEndpoint.class.getName());
+
+    public WebhookEndpoint() {
+        globalConfig = GlobalConfiguration.all().get(GlobalConfig.class);
+    }
+
+    @Override
+    public String getUrlName() {
+        String url = globalConfig.getSlackOutgoingWebhookURL();
+        if (url == null || url.equals(""))
+            return null;
+
+        return "/"+url;
+    }
+
+    @RequirePOST
+    public HttpResponse doIndex(StaplerRequest req) throws IOException,
+        ServletException {
+
+        if (globalConfig.getSlackOutgoingWebhookToken() == null ||
+            globalConfig.getSlackOutgoingWebhookToken().equals("")) {
+            return new JsonResponse(new SlackTextMessage("Slack token not set"),
+                StaplerResponse.SC_OK); 
+        }
+
+        SlackPostData data = new SlackPostData();
+        req.bindParameters(data);
+
+        if (!globalConfig.getSlackOutgoingWebhookToken().equals(data.getToken()))
+            return new JsonResponse(new SlackTextMessage("Invalid Slack token"),
+                StaplerResponse.SC_OK); 
+    
+        String commandText = data.getText();
+        if (commandText == null || commandText.isEmpty())
+            return new JsonResponse(new SlackTextMessage("Invalid command, text field required"),
+                StaplerResponse.SC_OK);
+
+        String triggerWord = data.getTrigger_word();
+        if (triggerWord == null || triggerWord.isEmpty())
+            return new JsonResponse(new SlackTextMessage("Invalid command, trigger_word field required"),
+                StaplerResponse.SC_OK);
+
+        //
+        // TODO: figure out better way of injecting slack user data into handler
+        //
+        this.slackUser = data.getUser_name();
+
+        CommandRouter<SlackTextMessage> router =
+            new CommandRouter<SlackTextMessage>();
+
+        try {
+            router.addRoute("^"+triggerWord+" list projects",
```

Comment:
Nice find.

The trigger word is set in Slack by the administrator, so the attacker would need admin access to create/use that trigger word. The trigger word in the routes is redundant however, so i'll remove that.